internal/provider/resource_tfe_organization_run_task.go

```diff
@@ -1,209 +1,272 @@
 // Copyright (c) HashiCorp, Inc.
 // SPDX-License-Identifier: MPL-2.0
 
-// NOTE: This is a legacy resource and should be migrated to the Plugin
-// Framework if substantial modifications are planned. See
-// docs/new-resources.md if planning to use this code as boilerplate for
-// a new resource.
-
 package provider
 
 import (
 	"context"
+	"errors"
 	"fmt"
-	"log"
-	"strings"
 
 	tfe "github.com/hashicorp/go-tfe"
-	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
-	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/booldefault"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/planmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/stringdefault"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/stringplanmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/schema/validator"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+
+	customValidators "github.com/hashicorp/terraform-provider-tfe/internal/provider/validators"
 )
 
-func resourceTFEOrganizationRunTask() *schema.Resource {
-	return &schema.Resource{
-		Create: resourceTFEOrganizationRunTaskCreate,
-		Read:   resourceTFEOrganizationRunTaskRead,
-		Delete: resourceTFEOrganizationRunTaskDelete,
-		Update: resourceTFEOrganizationRunTaskUpdate,
-		Importer: &schema.ResourceImporter{
-			StateContext: resourceTFEOrganizationRunTaskImporter,
-		},
+type resourceOrgRunTask struct {
+	config ConfiguredClient
+}
+
+func NewOrganizationRunTaskResource() resource.Resource {
+	return &resourceOrgRunTask{}
+}
+
+type modelTFEOrganizationRunTaskV0 struct {
+	Category     types.String `tfsdk:"category"`
+	Description  types.String `tfsdk:"description"`
+	Enabled      types.Bool   `tfsdk:"enabled"`
+	HMACKkey     types.String `tfsdk:"hmac_key"`
```

Comment:
Looks like a typo here

```suggestion
	HMACKey     types.String `tfsdk:"hmac_key"`
```

Reply:
Nice catch.  Thanks.